modules/auxiliary/gather/python_flask_cookie_signer.rb

```diff
@@ -45,54 +45,62 @@ def initialize(info = {})
       [
         Opt::RPORT(80),
         OptString.new('TARGETURI', [ true, 'URI to browse', '/']),
-        OptString.new('NEWCOOKIECONTENT', [ false, 'Content of a cookie to sign', '']),
-        OptString.new('SECRET', [ false, 'Content of a cookie to sign', '']),
+        OptString.new('NEWCOOKIECONTENT', [ false, 'Content of a cookie to sign', ''], conditions: %w[ACTION == Resign]),
+        OptString.new('SECRET', [ true, 'The key with which to sign the cookie', '']),
+      ]
+    )
+    register_advanced_options(
+      [
+        OptString.new('CookieName', [ true, 'The name of the session cookie', 'session' ]),
+        OptString.new('Salt', [ true, 'The salt to use for key derivation', Msf::Exploit::Remote::HTTP::FlaskUnsign::Session::DEFAULT_SALT ])
```

Comment:
These advanced options grant more flexibility to the user so if non-default values are in use, the module will still work. Some Flask extensions such as [Flask-Session](https://github.com/pallets-eco/flask-session/blob/09094d30cfa1aace46f93a192260940bf95cf97b/src/flask_session/sessions.py#L68) use a non-default salt.